src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -866,29 +866,33 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
   }
 
   def cudaTranspose[N:Numeric:Manifest](implicit __pos: SourceContext) = cudaGlobalFun {
-    (in: Rep[Array[N]], out: Rep[Array[N]]) =>
+    (in: Rep[Array[N]], out: Rep[Array[N]], n: Rep[Int], m: Rep[Int]) =>
       generate_comment("Cuda Coalesced Transpose")
-      (generate_comment("arg0: 2D Input Matrix (n x n) where n is a multiple of 32"))
-      (generate_comment("arg1: 2D Output Matrix (n x n) where n is a multiple of 32"))
+      generate_comment("arg0: 2D Input Matrix (n x m)")
+      generate_comment("arg1: 2D Output Transposed Matrix (m x n)")
+
       val tile = NewSharedArray[N](tileDim, tileDim + 1)
-      val x = blockIdxX * tileDim + threadIdxX
-      val y = blockIdxY * tileDim + threadIdxY
-      val width = gridDimX * tileDim
 
-      for(i <- (0 until (tileDim, blockRows)): Rep[Range]) {
-          tile(threadIdxY + i, threadIdxX) = in((y + i) * width + x)
+      for (i <- (0 until (tileDim, blockRows)): Rep[Range]) {
+        val x = blockIdxX * tileDim + threadIdxX
+        val y = blockIdxY * tileDim + threadIdxY + i
```

Comment:
could some of the computation be left before this for loop?

Reply:
done